arviz/plots/backends/matplotlib/lmplot.py

```diff
@@ -0,0 +1,44 @@
+"""Matplotlib plot linear regression figure."""
+import matplotlib.pyplot as plt
+import numpy as np
+
+from ...hdiplot import plot_hdi
+
+
+def plot_lm(x, y, y_model, y_ppc, num_pp_samples, kind_pp, kind_model):
+    """Matplotlib Linear Regression."""
+    _, ax = plt.subplots(1, 1, figsize=(6, 4), dpi=100)
+
+    # plot data
+    ax.plot(x, y, marker=".", color="C3", lw=0, zorder=10)
+
+    if y_ppc is not None:
+        arr = list(range(y_ppc.draw.size * y_ppc.chain.size))
+        slicer = np.random.choice(arr, size=num_pp_samples, replace=False)
+
+        if kind_pp == "samples":
+            y_ppc_ = y_ppc.stack(sample=("chain", "draw"))[..., slicer]
+            for i in range(num_pp_samples):
+                ax.plot(x, y_ppc_[..., i], marker=".", lw=0, alpha=0.1, color="C1")
+        else:
+            plot_hdi(x, y_ppc, color="C1")
+
+    # plot uncertainty in mean line
+    if y_model is not None:
+        arr = list(range(y_model.draw.size * y_ppc.chain.size))
+        slicer = np.random.choice(arr, size=num_pp_samples, replace=False)
+        if kind_model == "lines":
+            y_model_ = y_model.stack(sample=("chain", "draw"))[..., slicer]
+            for i in range(num_pp_samples):
+                ax.plot(x, y_model_[..., i], color="k", lw=0.5, alpha=0.5, zorder=9)
+        else:
+            plot_hdi(x, y_model, color="k", fill_kwargs={"alpha": 0.5, "lw": 0.5, "zorder": 9})
+
+        # plot mean line
+        y_model_mean_ = y_model.stack(sample=("chain", "draw"))[..., slicer].mean("sample")
+        ax.plot(x, y_model_mean_, color="y", lw=0.8, zorder=11)
+
+    for spine in ax.spines.values():
```

Comment:
I think this also needs some thinking when we want to do this and when not

Reply:
Yes , I have just taken it from from your function. I need to think about it because I have less idea where is it useful.